### PR TITLE
Implemented input validation for package name length

### LINKED ISSUE #1452: Do input validation on package name length
It looks like we're getting some API queries with a package name that is too long and results in an error:

```
google.cloud.ndb.exceptions.BadValueError: Indexed value project must be at most 1500 bytes

at ._validate ( /usr/local/lib/python3.11/site-packages/google/cloud/ndb/model.py:2926 )
at .call ( /usr/local/lib/python3.11/site-packages/google/cloud/ndb/model.py:1834 )
at ._call_shallow_validation ( /usr/local/lib/python3.11/site-packages/google/cloud/ndb/model.py:1762 )
at ._do_validate ( /usr/local/lib/python3.11/site-packages/google/cloud/ndb/model.py:1404 )
at ._comparison ( /usr/local/lib/python3.11/site-packages/google/cloud/ndb/model.py:1226 )
at .__eq__ ( /usr/local/lib/python3.11/site-packages/google/cloud/ndb/model.py:1238 )
at .query_by_version ( /osv/gcp/api/server.py:796 )
at ._advance_tasklet ( /usr/local/lib/python3.11/site-packages/google/cloud/ndb/tasklets.py:323 )
at .do_query ( /osv/gcp/api/server.py:487 )
at ._advance_tasklet ( /usr/local/lib/python3.11/site-packages/google/cloud/ndb/tasklets.py:319 )
at .check_success ( /usr/local/lib/python3.11/site-packages/google/cloud/ndb/tasklets.py:157 )
at .result ( /usr/local/lib/python3.11/site-packages/google/cloud/ndb/tasklets.py:210 )
at .QueryAffectedBatch ( /osv/gcp/api/server.py:173 )
at .wrapper ( /osv/gcp/api/server.py:84 )
at ._call_behavior ( /usr/local/lib/python3.11/site-packages/grpc/_server.py:494 )
```

### PR DESCRIPTION
**Pull request by masterismail**
I've implemented the input validation for the package name length. The package name will now be checked to ensure it does not exceed the maximum allowed length of 100 characters.

Please review the changes and let me know if you have any feedback or concerns. Once the changes are merged, this issue closes #1452

### CODE REVIEW

File: gcp/api/server.py [hunk 1]
```diff
@@ -66,6 +66,8 @@
 # This should match the number in the indexer
 _BUCKET_SIZE = 512
 
+MAX_PACKAGE_NAME_LENGTH = 100  # Maximum allowed package name length
```
Comment (andrewpollock) [inline gcp/api/server.py:69]: Based on the error message cited in #1452 I think 1500 is the 

```suggestion
MAX_PACKAGE_NAME_LENGTH = 1500  # Maximum allowed package name length
```

File: gcp/api/server.py [hunk 2]
```diff
@@ -795,6 +800,7 @@ def query_by_version(context: QueryContext,
                      version,
                      to_response: Callable = bug_to_response):
   """Query by (fuzzy) version."""
+  validate_package_name(package_name)
```
Comment (andrewpollock) [inline gcp/api/server.py:803]: https://github.com/google/osv.dev/blob/6246f3be6eaaee14f6616171ef7d133c4b3cc68a/gcp/api/server.py#L445 is probably a better place to do this validation, as it is

a) earlier in the call chain
b) will also catch it for `query_by_package()`, which needs this as well

**Review by andrewpollock — CHANGES_REQUESTED** [review]
Hello! Thanks for making this contribution, I've left some feedback.

**Comment by andrewpollock** [comment]
@masterismail do you plan on completing this, or should we close it out?

**Comment by masterismail** [comment]
hey @andrewpollock , apologise for the delay in getting back to you. I'll be updating the code with the requested changes asap.

**Comment by andrewpollock** [comment]
@masterismail there's some linter errors on your change:

```
gcp/api/server.py:440:0: W0311: Bad indentation. Found 4 spaces, expected 2 (bad-indentation)
gcp/api/server.py:441:0: W0311: Bad indentation. Found 8 spaces, expected 4 (bad-indentation)
gcp/api/server.py:442:0: C0303: Trailing whitespace (trailing-whitespace)
gcp/api/server.py:794:0: C0303: Trailing whitespace (trailing-whitespace)
```

Please run https://github.com/google/osv.dev/blob/master/tools/lint_and_format.sh on your changes prior to pushing them.

**Comment by andrewpollock** [comment]
@masterismail there's still some formatting errors with this change. Have you run https://github.com/google/osv.dev/blob/master/tools/lint_and_format.sh on it locally? If you run `yapf -i gcp/api/server.py` it should fix them.

**Comment by masterismail** [comment]
> @masterismail there's still some formatting errors with this change. Have you run https://github.com/google/osv.dev/blob/master/tools/lint_and_format.sh on it locally? If you run `yapf -i gcp/api/server.py` it should fix them.

yes, I have run "https://github.com/google/osv.dev/blob/master/tools/lint_and_format.sh " locally, this is what I am getting


Screenshot from 2023-08-08 16-52-25

Please let me know what exactly needs to fixed

**Comment by andrewpollock** [comment]
> this is what I am getting

Hi, have you set up your local development environment per https://github.com/google/osv.dev/blob/master/CONTRIBUTING.md#contributing-code?

Actually, reviewing that in conjunction with the output you've screenshotted, I can see that there's an undocumented requirement on having the Go toolchain installed. And `yapf`, which is why you didn't get locally notified about the formatting issues.

Are you as a contributor able to view https://github.com/google/osv.dev/actions/runs/5795924263/job/15708435878?pr=1455 ? That's showing the output of running `lint_and_format.sh` as part of our CI pipeline.

**Comment by masterismail** [comment]
> Are you as a contributor able to view https://github.com/google/osv.dev/actions/runs/5795924263/job/15708435878?pr=1455 ? That's showing the output of running `lint_and_format.sh` as part of our CI pipeline.

yes, i'm able to see those test fails

**Comment by masterismail** [comment]
> > this is what I am getting
> 
> Hi, have you set up your local development environment per https://github.com/google/osv.dev/blob/master/CONTRIBUTING.md#contributing-code?

yes, I did

**Comment by andrewpollock** [comment]
> yes, I did

Yes, it turns out there were a few things necessary that weren't explicitly in the documentation, see the #1542 for what else you need to have to be able to properly check your changes locally.

**Comment by andrewpollock** [comment]
> yes, i'm able to see those test fails

Cool, do you understand what you need to do to get your PR to pass, based on the failure?

**Comment by masterismail** [comment]
> > yes, i'm able to see those test fails
> 
> Cool, do you understand what you need to do to get your PR to pass, based on the failure?

I don't understand why its actually failing, it rates my code 10/10

**Comment by andrewpollock** [comment]
> I don't understand why its actually failing, it rates my code 10/10

`pylint` is happy with the change, `yapf` is not:

```
--- gcp/api/server.py	(original)
+++ gcp/api/server.py	(reformatted)
@@ -436,9 +436,12 @@
                                         empty_bucket_bitmap,
                                         len(version_query.file_hashes))
 
+
 def validate_package_name(package_name):
   if len(package_name) > MAX_PACKAGE_NAME_LENGTH:
     raise ValueError("Package name exceeds the maximum allowed length.")
+
+
 @ndb.tasklet
 def do_query(query, context: QueryContext, include_details=True):
   """Do a query."""
@@ -780,6 +783,7 @@
 
   return results, cursor
 
+
```

This is basically saying that you need to insert a blank line before your `validate_package_name` function and two blank lines after it (before the next function definition, `do_query`).

Take a moment to look at the formatting of the rest of the file, and you'll see that your change is inconsistent with the formatting of the file. You may also like to familiarise yourself with the [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html) more broadly, for future reference.

**Comment by andrewpollock** [comment]
@masterismail checking back in to see if you plan on completing this, or if we should close it?

**Comment by masterismail** [comment]
I've committed the changes as requested !

**Comment by andrewpollock** [comment]
@masterismail it looks like there's one last lint finding to address, you're close!

**Pull request closed without merging**